src/helics/common/addTargets.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright � 2017-2018,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC
+All rights reserved. See LICENSE file and DISCLAIMER for more details.
+*/
+#pragma once
+
+#include <string>
+#include <type_traits>
+
+namespace helics
+{
+template <typename Callable>
+void addTargets (const toml::Value &section, std::string targetName, Callable callback)
+{
+    // There should probably be a static_assert here but there isn't a nice type trait to check that
+    auto targets = section.find (targetName);
+    if (targets != nullptr)
+    {
+        if (targets->is<toml::Array> ())
+        {
+            auto &targetArray = targets->as<toml::Array> ();
+            for (const auto &target : targetArray)
+            {
+                callback (target.as<std::string> ());
+            }
+        }
+        else
+        {
+            callback (targets->as<std::string> ());
+        }
+    }
+    if (targetName.back () == 's')
+    {
+        targetName.pop_back ();
+        auto target = section.find (targetName);
+        if (target != nullptr)
+        {
+            callback (target->as<std::string> ());
+        }
+    }
+}
+
+template <typename Callable>
+void addTargets (const Json_helics::Value &section, std::string targetName, Callable callback)
```

Comment:
The jsoncpp library is included but since it has a cpp component it resulted in a number of symbol collisions when helics was used with other programs that used jsoncpp, so it needs to be wrapped in a different namespace.    I will migrate it to helics::Json namespace the next time I upgrade it as a related comment.    But for now everything related to jsoncpp is in json_helics namespace